docs/src/DeveloperGuide/WritingAdaptor.md

```diff
@@ -0,0 +1,264 @@
+upon configuring the profile. To this end, we need a configuration which allows
+the user to specify the message and we capture this configuration in a class
+which we name `HelloWorldConfig`:
+
+```c++
+using String = std::string;
+class HelloWorldConfig
+{
+public:
+  // ...
+private:
+  String message_{"Hello world"};
+};
+```
+
+We note the default value of our configuration is captured through the
+initialization of the class member. That is, if not overridden by the command
+line arguments, the message will be `"Hello world"`.
+
+To fulfil the concept of being a configuration, a configuration must implement a
+`setup` function taking a reference to a `ConfigurationManager` as its only
+argument. For our configuration, this looks like
+
+```c++
+class HelloWorldConfig
+{
+public:
+  void setup(ConfigurationManager &config)
+  {
+    config.setSectionName("Hello world configuration",
+                          "Demonstrating how configuration works.");
+    config.addParameter(message_, "message",
+                        "Message which is printed when setting the adaptor up.");
+  }
+
+  // ...
+};
+```
+
+The purpose of the `setup(config)` function is to inform the configuration
+manager about what the name of the configuration section and its description is
+as well as defining all settings and bind them to C++ variables. The benefit of
+this approach is that all configuration parameters for the adaptor will be
+available immediately after the adaptor is loaded by the tool.
+
+The final code to manage the configuration reads:
+
+```c++
+class HelloWorldConfig
+{
+public:
+  using String = std::string;
+
+  void setup(ConfigurationManager &config)
+  {
+    config.setSectionName("Hello world configuration",
+                          "Demonstrating how configuration works.");
+    config.addParameter(message_, "message",
+                        "Message which is printed when setting the adaptor up.");
+  }
+
+  String const& message() const
+  {
+    return message_;
+  }
+
+private:
+  String message_{"Hello world"};
+};
+```
+
+With the configuration in place, the next thing we concern ourselves with is
+loading the adaptor. This is the functionality that registers the configuration
+together with an ID and a profile setup function. In our case, the setup
+function should just print a message given a `HelloWorldConfig` instance. The
+corresponding adaptor registration reads:
+
+```c++
+extern "C" void loadAdaptor(IQirAdaptorFactory *generator)
+{
+  generator->registerAdaptorComponent<HelloWorldConfig>(
+      "adaptor.hello-world",
+      [](HelloWorldConfig const &cfg, IQirAdaptorFactory * /*generator*/, Profile & /*profile*/) {
+        std::cout << "Message: " << cfg.message() << std::endl;
+      });
+}
+```
+
+In this example, we will only concern ourselves with how to use the
+configuration and we will ignore `generator` and `profile` for now. The full
```

Comment:
Will `generator` and `profile` be covered in the second example? If so, it would be good to mention and link that here. If not, it would be good to add an example to cover these topics.